src/libcmd/installables.cc

```diff
@@ -102,6 +102,28 @@ MixFlakeOptions::MixFlakeOptions()
         }}
     });
 
+    addFlag({
+        .longName = "reference-lock-file",
```

Comment:
I'm not sure if there is a real use case for a separate `--reference-lock-file` and `--output-lock-file`. With a unified `--lock-file`, the caller can always write the input lock file to that location to be overwritten.

Reply:
> without having a checkout of the whole repository

Just for clarity: you mean checkout out the repository into a temporary directory because it will still land "checked out" in the nix store.

Reply:
Not necessarily in the future (#6530).